scheduling-api-http/src/integration-test/java/org/ow2/proactive/scheduling/api/GraphqlServiceIntegrationTest.java

```diff
@@ -671,6 +673,103 @@ public void testQueryTasksFilterByStatus() {
+    }
+
+    @Rollback
+    @Test
+    @Transactional
+    public void testQueryViewerJobs() {
+        addJobData(10);
+        JobData removedJob = createJobData("removed", "bobot", JobPriority.HIGH, "test", JobStatus.KILLED);
+        removedJob.setRemovedTime(14232323);
+        entityManager.persist(removedJob);
+
+        Map<String, Object> queryResult = executeGraphqlQuery("{ viewer { jobs  { edges { node { id owner } } } } }");
+
+        List<?> jobNodes = (List<?>) getField(queryResult, "data", "viewer", "jobs", "edges");
+        assertThat(jobNodes).hasSize(5);
+
+        jobNodes.forEach(jobNode -> assertThat(getField(jobNode, "node", "owner")).isEqualTo(CONTEXT_LOGIN));
+    }
+
+    @Rollback
+    @Test
+    @Transactional
+    public void testQueryViewerIncludeRemovedJobs() {
+        addJobData(10);
+        JobData removedJob = createJobData("removed", "bobot", JobPriority.HIGH, "test", JobStatus.KILLED);
+        removedJob.setRemovedTime(14232323);
+        entityManager.persist(removedJob);
+
+        Map<String, Object> queryResult = executeGraphqlQuery("{ viewer { jobs (filter: {excludeRemoved: false, status: KILLED}) { edges { node { id owner } } } } }");
+
+        List<?> jobNodes = (List<?>) getField(queryResult, "data", "viewer", "jobs", "edges");
+        assertThat(jobNodes).hasSize(1);
+
+        jobNodes.forEach(jobNode -> assertThat(getField(jobNode, "node", "owner")).isEqualTo(CONTEXT_LOGIN));
+    }
+
+    @Rollback
+    @Test
+    @Transactional
+    public void testQueryJobsFilterByVariables() {
+        addJobData(1);
+
+        JobData uselessJob = createJobData("Useless Toto Job", "ninipou", JobPriority.NORMAL, "", JobStatus.RUNNING);
+        entityManager.persist(uselessJob);
+        JobData importantJob = createJobData("Important Toto Job", "ninipou", JobPriority.HIGH, "", JobStatus.RUNNING);
+        entityManager.persist(importantJob);
+        Map<String, JobDataVariable> variables = new HashMap<>();
+        JobDataVariable cmdVariable = createJobDataVariable(importantJob, "cmd", "echo hello toto !");
+        variables.put("cmd", cmdVariable);
+        entityManager.persist(cmdVariable);
+        JobDataVariable argTotoVariable = createJobDataVariable(importantJob, "arg", "toto");
+        variables.put("arg", argTotoVariable);
+        entityManager.persist(argTotoVariable);
+        importantJob.setVariables(variables);
+
+        Map<String, Object> queryAllTotoJobsResult = executeGraphqlQuery("{ jobs (filter: [{variables: [{key: \"cmd\", value: \"%\"}]} {name: \"*Job\"}]) { totalCount edges { node { name } } } }");
+        assertThat((Integer) getField(queryAllTotoJobsResult, "data", "jobs", "totalCount")).isEqualTo(2);
+
+        Map<String, Object> queryImportantJobResult = executeGraphqlQuery("{ jobs (filter: {variables: [{key: \"cmd\", value: \"%\"}]}) { totalCount edges { node { name } } } }");
```

Comment:
it could be interesting to test filter both variables "arg" and "cmd"

Reply:
done !